Speed up Node refcounting via its EventTarget base class

cdumez opened the pull request:
#### ec525255894ef12c39407066bad4a315be7f9391
<pre>
Speed up Node refcounting via its EventTarget base class
<a href="https://bugs.webkit.org/show_bug.cgi?id=254113">https://bugs.webkit.org/show_bug.cgi?id=254113</a>

Reviewed by Darin Adler.

Speed up Node refcounting via its EventTarget base class by
avoiding virtual function calls.

* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTarget::ref): Deleted.
(WebCore::EventTarget::deref): Deleted.
* Source/WebCore/dom/Node.h:
(WebCore::EventTarget::ref):
(WebCore::EventTarget::deref):

Canonical link: <a href="https://commits.webkit.org/261849@main">https://commits.webkit.org/261849@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/955e1a7c5bd40dc01bc6f885980c8742b693ac68

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/113084 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/22217 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1750 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4855 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121547 "Failed to checkout and rebase branch from PR 11680") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/117189 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23587 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/13400 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/6093 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118871 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/17527 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100789 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/106166 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99481 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/1325 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/46554 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14519 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/1366 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/15234 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/10702 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20528 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53358 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8275 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/17080 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
Apparently you will have to figure out how to get `Node.h` included in the generated `JSTouchEvent.cpp` to land this.

cdumez committed:
Speed up Node refcounting via its EventTarget base class
https://bugs.webkit.org/show_bug.cgi?id=254113

Reviewed by Darin Adler.

Speed up Node refcounting via its EventTarget base class by
avoiding virtual function calls.

* Source/WebCore/dom/EventTarget.h:
(WebCore::EventTarget::ref): Deleted.
(WebCore::EventTarget::deref): Deleted.
* Source/WebCore/dom/Node.h:
(WebCore::EventTarget::ref):
(WebCore::EventTarget::deref):

Canonical link: https://commits.webkit.org/261849@main
